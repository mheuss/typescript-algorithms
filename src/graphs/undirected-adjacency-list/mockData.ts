export interface IStates {
  capital: string;
}

export interface IDistance {
  miles: number;
}
