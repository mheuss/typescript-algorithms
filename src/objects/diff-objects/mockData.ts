import { Banks } from "./mockTypes";

export const originalFile: Banks = [
  {
    bank_name: "Tait",
    city: "Saint Louis",
    routing_number: 316489490,
    state: "Missouri",
    zip: "63126",
  },
  {
    bank_name: "Merle",
    city: "Denver",
    routing_number: 732919062,
    state: "Colorado",
    zip: "80930",
  },
  {
    bank_name: "Francis",
    city: "Worcester",
    routing_number: 202463187,
    state: "Massachusetts",
    zip: "01605",
  },
  {
    bank_name: "Kirsteni",
    city: "Seattle",
    routing_number: 279027748,
    state: "Washington",
    zip: "98104",
  },
  {
    bank_name: "Danielle",
    city: "Midland",
    routing_number: 151532905,
    state: "Texas",
    zip: "78703",
  },
  {
    bank_name: "Shelby",
    city: "Shreveport",
    routing_number: 149772497,
    state: "Louisiana",
    zip: "71130",
  },
  {
    bank_name: "Carolin",
    city: "Saint Paul",
    routing_number: 617706482,
    state: "Minnesota",
    zip: "55103",
  },
  {
    bank_name: "Bili",
    city: "Saint Paul",
    routing_number: 527362918,
    state: "Minnesota",
    zip: "55146",
  },
  {
    bank_name: "Karilynn",
    city: "Honolulu",
    routing_number: 224887802,
    state: "Hawaii",
    zip: "96850",
  },
  {
    bank_name: "Ward",
    city: "Chicago",
    routing_number: 331421157,
    state: "Illinois",
    zip: "61656",
  },
  {
    bank_name: "Jolynn",
    city: "Houston",
    routing_number: 920233367,
    state: "Texas",
    zip: "77010",
  },
  {
    bank_name: "Dorelia",
    city: "Memphis",
    routing_number: 547444708,
    state: "Tennessee",
    zip: "38131",
  },
  {
    bank_name: "Gwenni",
    city: "Wichita",
    routing_number: 701886244,
    state: "Kansas",
    zip: "67215",
  },
  {
    bank_name: "Jaime",
    city: "Indianapolis",
    routing_number: 760086826,
    state: "Indiana",
    zip: "46295",
  },
  {
    bank_name: "Phillipe",
    city: "Largo",
    routing_number: 145543598,
    state: "Florida",
    zip: "34643",
  },
  {
    bank_name: "Samson",
    city: "Jamaica",
    routing_number: 593655726,
    state: "New York",
    zip: "11436",
  },
  {
    bank_name: "Delmer",
    city: "Denver",
    routing_number: 423427452,
    state: "Colorado",
    zip: "80270",
  },
  {
    bank_name: "Walsh",
    city: "Oklahoma City",
    routing_number: 228880440,
    state: "Oklahoma",
    zip: "73135",
  },
  {
    bank_name: "Trula",
    city: "Bozeman",
    routing_number: 811244802,
    state: "Montana",
    zip: "59771",
  },
  {
    bank_name: "Viv",
    city: "Albuquerque",
    routing_number: 692682035,
    state: "New Mexico",
    zip: "87180",
  },
  {
    bank_name: "Dollie",
    city: "San Antonio",
    routing_number: 707249543,
    state: "Texas",
    zip: "78291",
  },
  {
    bank_name: "Bernardo",
    city: "Richmond",
    routing_number: 859971484,
    state: "Virginia",
    zip: "23293",
  },
  {
    bank_name: "Rosita",
    city: "Albany",
    routing_number: 675519439,
    state: "New York",
    zip: "12237",
  },
  {
    bank_name: "Erin",
    city: "Denver",
    routing_number: 498873111,
    state: "Colorado",
    zip: "80209",
  },
  {
    bank_name: "Melba",
    city: "Clearwater",
    routing_number: 535335163,
    state: "Florida",
    zip: "33758",
  },
  {
    bank_name: "Sandor",
    city: "New Orleans",
    routing_number: 916490031,
    state: "Louisiana",
    zip: "70183",
  },
  {
    bank_name: "Petronella",
    city: "Omaha",
    routing_number: 747464677,
    state: "Nebraska",
    zip: "68134",
  },
  {
    bank_name: "Eunice",
    city: "Lansing",
    routing_number: 407981014,
    state: "Michigan",
    zip: "48930",
  },
  {
    bank_name: "Mannie",
    city: "New York City",
    routing_number: 304952129,
    state: "New York",
    zip: "10184",
  },
  {
    bank_name: "Avery",
    city: "Tulsa",
    routing_number: 936501142,
    state: "Oklahoma",
    zip: "74126",
  },
  {
    bank_name: "Valene",
    city: "Fairbanks",
    routing_number: 173060444,
    state: "Alaska",
    zip: "99790",
  },
  {
    bank_name: "Janifer",
    city: "Lubbock",
    routing_number: 530484329,
    state: "Texas",
    zip: "79491",
  },
  {
    bank_name: "Griffin",
    city: "Sacramento",
    routing_number: 288115184,
    state: "California",
    zip: "94297",
  },
  {
    bank_name: "Lilly",
    city: "Kissimmee",
    routing_number: 842286875,
    state: "Florida",
    zip: "34745",
  },
  {
    bank_name: "Yoko",
    city: "Washington",
    routing_number: 184481379,
    state: "District of Columbia",
    zip: "20337",
  },
  {
    bank_name: "Cherlyn",
    city: "North Little Rock",
    routing_number: 514038385,
    state: "Arkansas",
    zip: "72118",
  },
  {
    bank_name: "Pen",
    city: "Los Angeles",
    routing_number: 937516362,
    state: "California",
    zip: "90101",
  },
  {
    bank_name: "Court",
    city: "Washington",
    routing_number: 426945562,
    state: "District of Columbia",
    zip: "20503",
  },
  {
    bank_name: "Marcy",
    city: "Norfolk",
    routing_number: 154955464,
    state: "Virginia",
    zip: "23520",
  },
  {
    bank_name: "Kizzie",
    city: "Austin",
    routing_number: 588297101,
    state: "Texas",
    zip: "78732",
  },
  {
    bank_name: "Marlane",
    city: "Peoria",
    routing_number: 312666186,
    state: "Illinois",
    zip: "61651",
  },
  {
    bank_name: "Parker",
    city: "Jackson",
    routing_number: 473229698,
    state: "Mississippi",
    zip: "39282",
  },
  {
    bank_name: "Kylila",
    city: "Birmingham",
    routing_number: 310231910,
    state: "Alabama",
    zip: "35295",
  },
  {
    bank_name: "Harmony",
    city: "Charlotte",
    routing_number: 270117236,
    state: "North Carolina",
    zip: "28210",
  },
  {
    bank_name: "Dehlia",
    city: "Lansing",
    routing_number: 713767044,
    state: "Michigan",
    zip: "48901",
  },
];

export const newFile: Banks = [
  {
    bank_name: "Tait",
    city: "Saint Louis",
    routing_number: 316489490,
    state: "Missouri",
    zip: "63126",
  },
  {
    bank_name: "Merle",
    city: "Colorado Springs",
    routing_number: 732919062,
    state: "Colorado",
    zip: "80930",
  },
  {
    bank_name: "Francis",
    city: "Worcester",
    routing_number: 202463187,
    state: "Massachusetts",
    zip: "01605",
  },
  {
    bank_name: "Kirsteni",
    city: "Seattle",
    routing_number: 279027748,
    state: "Washington",
    zip: "98104",
  },
  {
    bank_name: "Danielle",
    city: "Austin",
    routing_number: 151532905,
    state: "Texas",
    zip: "78703",
  },
  {
    bank_name: "Shelby",
    city: "Shreveport",
    routing_number: 149772497,
    state: "Louisiana",
    zip: "71130",
  },
  {
    bank_name: "Carolin",
    city: "Saint Paul",
    routing_number: 617706482,
    state: "Minnesota",
    zip: "55103",
  },
  {
    bank_name: "Bili",
    city: "Saint Paul",
    routing_number: 527362918,
    state: "Minnesota",
    zip: "55146",
  },
  {
    bank_name: "Karilynn",
    city: "Honolulu",
    routing_number: 224887802,
    state: "Hawaii",
    zip: "96850",
  },
  {
    bank_name: "Ward",
    city: "Peoria",
    routing_number: 331421157,
    state: "Illinois",
    zip: "61656",
  },
  {
    bank_name: "Samson",
    city: "Jamaica",
    routing_number: 593655726,
    state: "New York",
    zip: "11436",
  },
  {
    bank_name: "Delmer",
    city: "Denver",
    routing_number: 423427452,
    state: "Colorado",
    zip: "80270",
  },
  {
    bank_name: "Walsh",
    city: "Oklahoma City",
    routing_number: 228880440,
    state: "Oklahoma",
    zip: "73135",
  },
  {
    bank_name: "Trula",
    city: "Bozeman",
    routing_number: 811244802,
    state: "Montana",
    zip: "59771",
  },
  {
    bank_name: "Viv",
    city: "Albuquerque",
    routing_number: 692682035,
    state: "New Mexico",
    zip: "87180",
  },
  {
    bank_name: "Dollie",
    city: "San Antonio",
    routing_number: 707249543,
    state: "Texas",
    zip: "78291",
  },
  {
    bank_name: "Bernardo",
    city: "Richmond",
    routing_number: 859971484,
    state: "Virginia",
    zip: "23293",
  },
  {
    bank_name: "Rosita",
    city: "Albany",
    routing_number: 675519439,
    state: "New York",
    zip: "12237",
  },
  {
    bank_name: "Erin",
    city: "Denver",
    routing_number: 498873111,
    state: "Colorado",
    zip: "80209",
  },
  {
    bank_name: "Melba",
    city: "Clearwater",
    routing_number: 535335163,
    state: "Florida",
    zip: "33758",
  },
  {
    bank_name: "Sandor",
    city: "New Orleans",
    routing_number: 916490031,
    state: "Louisiana",
    zip: "70183",
  },
  {
    bank_name: "Petronella",
    city: "Omaha",
    routing_number: 747464677,
    state: "Nebraska",
    zip: "68134",
  },
  {
    bank_name: "Eunice",
    city: "Lansing",
    routing_number: 407981014,
    state: "Michigan",
    zip: "48930",
  },
  {
    bank_name: "Mannie",
    city: "New York City",
    routing_number: 304952129,
    state: "New York",
    zip: "10184",
  },
  {
    bank_name: "Avery",
    city: "Tulsa",
    routing_number: 936501142,
    state: "Oklahoma",
    zip: "74126",
  },
  {
    bank_name: "Valene",
    city: "Fairbanks",
    routing_number: 173060444,
    state: "Alaska",
    zip: "99790",
  },
  {
    bank_name: "Janifer",
    city: "Lubbock",
    routing_number: 530484329,
    state: "Texas",
    zip: "79491",
  },
  {
    bank_name: "Griffin",
    city: "Sacramento",
    routing_number: 288115184,
    state: "California",
    zip: "94297",
  },
  {
    bank_name: "Lilly",
    city: "Kissimmee",
    routing_number: 842286875,
    state: "Florida",
    zip: "34745",
  },
  {
    bank_name: "Yoko",
    city: "Washington",
    routing_number: 184481379,
    state: "District of Columbia",
    zip: "20337",
  },
  {
    bank_name: "Cherlyn",
    city: "North Little Rock",
    routing_number: 514038385,
    state: "Arkansas",
    zip: "72118",
  },
  {
    bank_name: "Pen",
    city: "Los Angeles",
    routing_number: 937516362,
    state: "California",
    zip: "90101",
  },
  {
    bank_name: "Court",
    city: "Washington",
    routing_number: 426945562,
    state: "District of Columbia",
    zip: "20503",
  },
  {
    bank_name: "Marcy",
    city: "Norfolk",
    routing_number: 154955464,
    state: "Virginia",
    zip: "23520",
  },
  {
    bank_name: "Kizzie",
    city: "Austin",
    routing_number: 588297101,
    state: "Texas",
    zip: "78732",
  },
  {
    bank_name: "Marlane",
    city: "Peoria",
    routing_number: 312666186,
    state: "Illinois",
    zip: "61651",
  },
  {
    bank_name: "Parker",
    city: "Jackson",
    routing_number: 473229698,
    state: "Mississippi",
    zip: "39282",
  },
  {
    bank_name: "Kylila",
    city: "Birmingham",
    routing_number: 310231910,
    state: "Alabama",
    zip: "35295",
  },
  {
    bank_name: "Harmony",
    city: "Charlotte",
    routing_number: 270117236,
    state: "North Carolina",
    zip: "28210",
  },
  {
    bank_name: "Dehlia",
    city: "Lansing",
    routing_number: 713767044,
    state: "Michigan",
    zip: "48901",
  },
  {
    bank_name: "Andres",
    city: "Flushing",
    routing_number: 836255617,
    state: "New York",
    zip: "11388",
  },
  {
    bank_name: "Stevie",
    city: "Lincoln",
    routing_number: 641717493,
    state: "Nebraska",
    zip: "68583",
  },
  {
    bank_name: "Alicia",
    city: "Springfield",
    routing_number: 812862008,
    state: "Missouri",
    zip: "65810",
  },
  {
    bank_name: "Sharia",
    city: "Abilene",
    routing_number: 807061088,
    state: "Texas",
    zip: "79605",
  },
  {
    bank_name: "Rocky",
    city: "Los Angeles",
    routing_number: 366140410,
    state: "California",
    zip: "90005",
  },
];
