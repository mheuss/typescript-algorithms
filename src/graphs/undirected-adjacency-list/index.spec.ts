import { UndirectedAdjacencyListGraph } from "./index";
import { IDistance, IStates } from "./mockData";
import {
  IShortestPathPayload,
  ITraversalPayload,
  ShortestPathCallback,
} from "./types";

describe("Undirected Adjacency Graph Unit tests", () => {
  describe("Dijkstra's Shortest Path", () => {
    const callback: ShortestPathCallback<IStates, IDistance> = (
      vertexAndEdge: IShortestPathPayload<IStates, IDistance>
    ) => {
      return vertexAndEdge.edge.payload.miles;
    };
    it("Should find the shortest path from Augusta to Albany", () => {
      const graph = setupGraphs();

      expect(graph.dijkstraShortestPath("Maine", "New York", callback)).toEqual(
        ["New Hampshire", "Vermont", "New York"]
      );
      expect(
        graph.dijkstraShortestPath("Massachusetts", "New York", callback)
      ).toEqual(["Connecticut", "New York"]);
    });

    it("Should return [] if there is no valid path - Connected to unconnected", () => {
      const graph = setupGraphs();
      graph.addVertex("Georgia", { capital: "Atlanta" });
      expect(graph.dijkstraShortestPath("Maine", "Georgia", callback)).toEqual(
        []
      );
    });

    it("Should return [] if there is no valid path - unconnected to connected", () => {
      const graph = setupGraphs();
      graph.addVertex("Georgia", { capital: "Atlanta" });
      expect(graph.dijkstraShortestPath("Georgia", "Maine", callback)).toEqual(
        []
      );
    });

    it("Should return empty array if asked to path for itself", () => {
      const graph = setupGraphs();
      expect(graph.dijkstraShortestPath("Maine", "Maine", callback)).toEqual(
        []
      );
    });

    it("Should throw if given nonexistant vertices", () => {
      const graph = setupGraphs();
      expect(() =>
        graph.dijkstraShortestPath("Ramsflat", "Maine", callback)
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        graph.dijkstraShortestPath("Maine", "Farblebutt", callback)
      ).toThrowErrorMatchingSnapshot();
    });
  });

  it("Should add a vertex when asked", () => {
    const graph = new UndirectedAdjacencyListGraph<string, boolean>();
    graph.addVertex("New Vertex", "Molly Coddle");
    expect(graph.getAdjacencyList()).toEqual({
      "New Vertex": { edges: {}, name: "New Vertex", payload: "Molly Coddle" },
    });
  });

  describe("Fetch Data from graph", () => {
    it("Should give me the payload of an item in a graph", () => {
      const graph = setupGraphs();
      expect(graph.getVertex("Massachusetts")).toEqual({ capital: "Boston" });
    });

    it("Should give me the payload of an vertex between two points in a graph", () => {
      const graph = setupGraphs();
      expect(graph.getEdge("Massachusetts", "New Hampshire")).toEqual({
        miles: 164,
      });
    });

    it("Should give me all the vertexes a single vertex is connected to", () => {
      const graph = setupGraphs();
      expect(graph.getEdges("Massachusetts")).toEqual([
        "Connecticut",
        "New Hampshire",
        "Vermont",
        "Rhode Island",
      ]);
    });

    it("Should return an empty array if we are trying to get edges from nothing", () => {
      const graph = setupGraphs();
      expect(graph.getEdges("Splindeforuos")).toEqual([]);
    });
  });

  describe("Add edges", () => {
    it("Should add an edge between vertices", () => {
      const graph = new UndirectedAdjacencyListGraph<boolean, boolean>();
      graph.addVertex("Point A", true);
      graph.addVertex("Point B", false);
      graph.addEdgeWithSeparatePayloads(
        { name: "Point A", payload: true },
        { name: "Point B", payload: false }
      );
      expect(graph.getAdjacencyList()).toEqual({
        "Point A": {
          edges: { "Point B": { name: "Point B", payload: true } },
          name: "Point A",
          payload: true,
        },
        "Point B": {
          edges: { "Point A": { name: "Point A", payload: false } },
          name: "Point B",
          payload: false,
        },
      });
    });

    it("Should throw if we try to add to a non existent vertex", () => {
      const graph = new UndirectedAdjacencyListGraph<boolean, boolean>();
      graph.addVertex("Point A", true);
      graph.addVertex("Point B", false);

      expect(() =>
        graph.addEdgeWithSeparatePayloads(
          { name: "Point A", payload: true },
          { name: "Point C", payload: true }
        )
      ).toThrowErrorMatchingSnapshot();
    });
  });

  it("Should add an edge between vertices with a shared payload", () => {
    const graph = new UndirectedAdjacencyListGraph<boolean, boolean>();
    graph.addVertex("Point A", true);
    graph.addVertex("Point B", false);
    graph.addEdgeWithSharedPayload("Point A", "Point B", false);
    expect(graph.getAdjacencyList()).toEqual({
      "Point A": {
        edges: { "Point B": { name: "Point B", payload: false } },
        name: "Point A",
        payload: true,
      },
      "Point B": {
        edges: { "Point A": { name: "Point A", payload: false } },
        name: "Point B",
        payload: false,
      },
    });
  });

  it("Should throw if we try to add to a non existent vertex", () => {
    const graph = new UndirectedAdjacencyListGraph<boolean, boolean>();
    graph.addVertex("Point A", true);
    graph.addVertex("Point B", false);

    expect(() =>
      graph.addEdgeWithSharedPayload("Point C", "Point B", false)
    ).toThrowErrorMatchingSnapshot();
  });
});

describe("Removal", () => {
  let graph: UndirectedAdjacencyListGraph<boolean, boolean>;

  beforeEach(() => {
    graph = new UndirectedAdjacencyListGraph<boolean, boolean>();
    graph.addVertex("Point A", true);
    graph.addVertex("Point B", true);
    graph.addVertex("Point C", false);
    graph.addVertex("Point D", false);
    graph.addEdgeWithSeparatePayloads(
      { name: "Point A", payload: true },
      { name: "Point D", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point A", payload: true },
      { name: "Point C", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point B", payload: true },
      { name: "Point D", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point B", payload: true },
      { name: "Point C", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point C", payload: true },
      { name: "Point A", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point C", payload: true },
      { name: "Point B", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point D", payload: true },
      { name: "Point A", payload: false }
    );
    graph.addEdgeWithSeparatePayloads(
      { name: "Point D", payload: true },
      { name: "Point B", payload: false }
    );
  });

  it("should remove an edge", () => {
    graph.removeEdge("Point A", "Point D");
    expect(graph.getAdjacencyList()).toMatchSnapshot();
  });

  it("Should remove a vertex", () => {
    graph.removeVertex("Point A");
    expect(graph.getAdjacencyList()).toMatchSnapshot();
  });
});

describe("Traversal", () => {
  let graph: UndirectedAdjacencyListGraph<IStates, IDistance>;
  beforeEach(() => {
    graph = setupGraphs();
  });
  it("Should do a Depth First Traversal", () => {
    const traversalAccumulator: string[] = [];
    const traversalCallback = (
      vertexAndEdge: ITraversalPayload<IStates, IDistance>
    ) => {
      if (vertexAndEdge.edge === undefined) {
        traversalAccumulator.push(`We are at ${vertexAndEdge.vertex.name}`);
        return;
      }
      traversalAccumulator.push(
        `We are at ${vertexAndEdge.vertex.name} and we came from ${vertexAndEdge.edge.name}`
      );
    };

    graph.depthFirstTraversal("Maine", traversalCallback);
    expect(traversalAccumulator).toMatchSnapshot();
  });

  it("Should not do a Depth First Traversal if we do not give it a valid starting point", () => {
    const traversalCallback = () => {
      return;
    };

    expect(() => {
      graph.depthFirstTraversal("Splasndis", traversalCallback);
    }).toThrowErrorMatchingSnapshot();
  });

  it("Should do a Breadth First Traversal", () => {
    const traversalAccumulator: string[] = [];
    const traversalCallback = (
      vertexAndEdge: ITraversalPayload<IStates, IDistance>
    ) => {
      if (vertexAndEdge.edge === undefined) {
        traversalAccumulator.push(`We are at ${vertexAndEdge.vertex.name}`);
        return;
      }
      traversalAccumulator.push(
        `We are at ${vertexAndEdge.vertex.name} and we came from ${vertexAndEdge.edge.name}`
      );
    };

    graph.breadthFirstTraversal("Maine", traversalCallback);
    expect(traversalAccumulator).toMatchSnapshot();
  });

  it("Should not do a Breadth First Traversal if we do not give it a valid starting point", () => {
    const traversalCallback = () => {
      return;
    };

    expect(() => {
      graph.breadthFirstTraversal("Splasndis", traversalCallback);
    }).toThrowErrorMatchingSnapshot();
  });
});

function setupGraphs() {
  const graph = new UndirectedAdjacencyListGraph<IStates, IDistance>();

  graph.addVertex("Connecticut", { capital: "Hartford" });
  graph.addVertex("Maine", { capital: "Augusta" });
  graph.addVertex("Massachusetts", { capital: "Boston" });
  graph.addVertex("New Hampshire", { capital: "Concord" });
  graph.addVertex("New York", { capital: "Albany" });
  graph.addVertex("Rhode Island", { capital: "Providence" });
  graph.addVertex("Vermont", { capital: "Montpelier" });

  graph.addEdgeWithSharedPayload("Connecticut", "Rhode Island", {
    miles: 87,
  });
  graph.addEdgeWithSharedPayload("Connecticut", "New York", { miles: 163 });
  graph.addEdgeWithSharedPayload("Connecticut", "Massachusetts", {
    miles: 50,
  });

  graph.addEdgeWithSharedPayload("Maine", "New Hampshire", { miles: 164 });

  graph.addEdgeWithSharedPayload("Massachusetts", "New Hampshire", {
    miles: 164,
  });
  graph.addEdgeWithSharedPayload("Massachusetts", "Vermont", { miles: 180 });
  graph.addEdgeWithSharedPayload("Massachusetts", "Rhode Island", {
    miles: 50,
  });

  graph.addEdgeWithSharedPayload("New Hampshire", "Vermont", { miles: 116 });

  graph.addEdgeWithSharedPayload("New York", "Vermont", { miles: 158 });

  return graph;
}
