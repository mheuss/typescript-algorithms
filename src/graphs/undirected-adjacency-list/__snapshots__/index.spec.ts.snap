// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Removal Should remove a vertex 1`] = `
Object {
  "Point B": Object {
    "edges": Object {
      "Point C": Object {
        "name": "Point C",
        "payload": true,
      },
      "Point D": Object {
        "name": "Point D",
        "payload": true,
      },
    },
    "name": "Point B",
    "payload": true,
  },
  "Point C": Object {
    "edges": Object {
      "Point B": Object {
        "name": "Point B",
        "payload": false,
      },
    },
    "name": "Point C",
    "payload": false,
  },
  "Point D": Object {
    "edges": Object {
      "Point B": Object {
        "name": "Point B",
        "payload": false,
      },
    },
    "name": "Point D",
    "payload": false,
  },
}
`;

exports[`Removal should remove an edge 1`] = `
Object {
  "Point A": Object {
    "edges": Object {
      "Point C": Object {
        "name": "Point C",
        "payload": true,
      },
    },
    "name": "Point A",
    "payload": true,
  },
  "Point B": Object {
    "edges": Object {
      "Point C": Object {
        "name": "Point C",
        "payload": true,
      },
      "Point D": Object {
        "name": "Point D",
        "payload": true,
      },
    },
    "name": "Point B",
    "payload": true,
  },
  "Point C": Object {
    "edges": Object {
      "Point A": Object {
        "name": "Point A",
        "payload": false,
      },
      "Point B": Object {
        "name": "Point B",
        "payload": false,
      },
    },
    "name": "Point C",
    "payload": false,
  },
  "Point D": Object {
    "edges": Object {
      "Point B": Object {
        "name": "Point B",
        "payload": false,
      },
    },
    "name": "Point D",
    "payload": false,
  },
}
`;

exports[`Traversal Should do a Depth First Traversal 1`] = `
Array [
  "We are at Maine",
  "We are at New Hampshire and we came from Maine",
  "We are at Massachusetts and we came from New Hampshire",
  "We are at Connecticut and we came from Massachusetts",
  "We are at Rhode Island and we came from Connecticut",
  "We are at New York and we came from Connecticut",
  "We are at Vermont and we came from New York",
]
`;

exports[`Undirected Adjacency Graph Unit tests Add edges Should throw if we try to add to a non existent vertex 1`] = `"One or both of these vertices don't exist Point A / Point C"`;

exports[`Undirected Adjacency Graph Unit tests Should throw if we try to add to a non existent vertex 1`] = `"One or both of these vertices don't exist Point C / Point B"`;
